framework/helpers/BaseArrayHelper.php

```diff
@@ -327,7 +327,12 @@ public static function setValue(&$array, $path, $value)
      */
     public static function remove(&$array, $key, $default = null)
     {
-        if (is_array($array) && (isset($array[$key]) || array_key_exists($key, $array))) {
+        // This check can be removed when the minimum PHP version is >= 8.1 (Yii2.2)
```

Comment:
Can this comment be added something that is actually detectable using automated means like TODO or deprecation or open ticket. I suggest an open ticket to track it? I'm afraid it might escapes the eyes of devs for a good time in Yii2.2.